lib/api/apiUtils/bucket/models/ReplicationConfiguration.js

```diff
@@ -0,0 +1,307 @@
+        if (!Rule || Rule.length < 1) {
+            return errors.MalformedXML;
+        }
+        if (Rule.length > 1000) {
+            return errors.InvalidRequest.customizeDescription(
+                'Number of defined replication rules cannot exceed 1000');
+        }
+        const err = this._parseEachRule(Rule);
+        if (err) {
+            return err;
+        }
+        this.replicationConfiguration.rules = this._rules;
+        return undefined;
+    }
+
+    /**
+     * Check that each rule in the `Rules` property array is valid
+     * @param {array} rules - The rule array from this._parsedXML
+     * @return {undefined}
+     */
+    _parseEachRule(rules) {
+        const rulesArr = [];
+        for (let i = 0; i < rules.length; i++) {
+            const err =
+                this._parseStatus(rules[i]) || this._parsePrefix(rules[i]) ||
+                this._parseID(rules[i]) || this._parseDestination(rules[i]);
+            if (err) {
+                return err;
+            }
+            rulesArr.push(this._getRuleObject(rules[i]));
+        }
+        this._rules = rulesArr;
+        return undefined;
+    }
+
+    /**
+     * Check that the `Status` property is valid
+     * @param {object} rule - The rule object from this._parsedXML
+     * @return {undefined}
+     */
+    _parseStatus(rule) {
+        const status = rule.Status && rule.Status[0];
+        if (!status || !['Enabled', 'Disabled'].includes(status)) {
+            return errors.MalformedXML;
+        }
+        return undefined;
+    }
+
+    /**
+     * Check that the `Prefix` property is valid
+     * @param {object} rule - The rule object from this._parsedXML
+     * @return {undefined}
+     */
+    _parsePrefix(rule) {
+        const prefix = rule.Prefix && (rule.Prefix[0] || rule.Prefix[0] === '');
+        if (!prefix && prefix !== '') {
+            return errors.MalformedXML;
+        }
+        if (prefix.length > 1024) {
+            return errors.InvalidArgument.customizeDescription('Rule prefix ' +
+                'cannot be longer than maximum allowed key length of 1024');
+        }
+        // Each Prefix in a list of rules must not overlap. For example, two
+        // prefixes 'TaxDocs' and 'TaxDocs/2015' are overlapping.
+        for (let i = 0; i < this._configPrefixes.length; i++) {
+            const usedPrefix = this._configPrefixes[i];
+            if (prefix.startsWith(usedPrefix)) {
+                return errors.InvalidRequest.customizeDescription('Found ' +
+                    `overlapping prefixes '${usedPrefix}' and '${prefix}'`);
+            }
+        }
+        this._configPrefixes.push(prefix);
+        return undefined;
+    }
+
+    /**
+     * Check that the `ID` property is valid
+     * @param {object} rule - The rule object from this._parsedXML
+     * @return {undefined}
+     */
+    _parseID(rule) {
+        const id = rule.ID && rule.ID[0];
+        if (id && id.length > 255) {
```

Comment:
Let's declare this as a named constant `const RULE_ID_LIMIT = 255;` and use it here.